src/core/structured_config.py

```diff
@@ -224,6 +227,18 @@ def profile_values(cls, value: str) -> str | None:
 
         return value
 
+    @validator("expose_external")
+    @classmethod
+    def expose_external_validator(cls, value: str) -> str | None:
+        """Check expose-external config option is only used on Kubernetes charm."""
+        if SUBSTRATE == "vm" and value:
+            raise ValueError("Value not permitted on VM charm")
+
+        if value == "none":
+            return
```

Comment:
**question**: Should we invert those two conditions? With this flow, we do not allow VM to pass `"none"`. Or will we remove the default value in the VM `config.yaml` file, e.g.
```yaml
  expose-external:
    description: "String to determine how to expose the Kafka cluster externally from the Kubernetes cluster. Possible values: 'nodeport', 'none'"
    type: string
    default: 
```

instead of 
```yaml
  expose-external:
    description: "String to determine how to expose the Kafka cluster externally from the Kubernetes cluster. Possible values: 'nodeport', 'none'"
    type: string
    default: "none"
```
?

Reply:
Ah shit, VERY good point. I don't have a good idea here. Maybe we specifically change the default value on the VM charm, what do you think?

Reply:
I think it's ok to have different defaults between the two charms right now. But I also think this raises a bigger question: would it be ok to just ignore this config option in VM (and eventually log that we ignore this option, for ~slightly better user XP)?

Because when (if?) we merge the two code bases in one single fit-all charm, this will be a problem

Reply:
I suppose I could change the validator to also return `none` if SUBSTRATE == "vm". Could do that now, what do you think?

Reply:
I would prefer that, we can always go back to blocking the charm if needed

Reply:
Uhm, I don't think we should expose the config option on VMs given it does not make much sense there: thus, I would remove the expose-external option from config.yaml on VM charms. It is still fine to have that key in the code (such that we can maximise the code sharing) and hard code the attribute in VM on charm.py or somewhere where it is convenient on VM.

Reply:
We're copying `structured_config`, I can check if it handles just not passing that config option?